proto-lens/src/Data/ProtoLens/Encoding.hs

```diff
@@ -53,20 +53,23 @@ parseMessage :: forall msg . Message msg => Parser () -> Parser msg
 parseMessage end = do
     (msg, unsetFields) <- loop def requiredFields
     if Map.null unsetFields
-        then return $ reverseRepeatedFields fields msg
+        then return $ over unknownFields reverse
+                    $ reverseRepeatedFields fields msg
         else fail $ "Missing required fields "
                         ++ show (map fieldDescriptorName
                                     $ Map.elems $ unsetFields)
   where
     fields = fieldsByTag descriptor
+    addUnknown :: TaggedValue -> msg -> msg
+    addUnknown !f = over unknownFields (\(!xs) -> f : xs)
```

Comment:
this might be cleaner as `over unknownFields (f :)`

Reply:
It needs the bang pattern or else horrible space leaks happen.  I've sent out #130 to add a helper function and use it elsewhere in this file; once that's in I'll use it here too.

Reply:
ahh, missed the bang. makes sense